### ISSUE TITLE
Respect Order of Color Domain

### ISSUE THREAD

**Issue by mtanco** [h2oai/wave#1269]
#### Is your feature request related to a problem? Please describe
In the below example you can see that the order of the `color_range` and `color_domain` categories are such that `A` is blue and `B` is red. However, colors from the `color_range` list are assigned to the values of the `color` column based on the order that they appear in the data object. So in our key the fields and in the "wrong" order and the categories don't have the colors we would expect. 

```python
    # Plot two numeric columns by each other and color based on a third, categorical column
    q.page['scatter'] = ui.plot_card(
        box='1 1 4 5',
        title='Scatter Plot from Dataframe',
        data=data(
            fields=["x_value", "y_value", "category"],
            rows=[
                [10, 100, 'B'],
                [100, 10, 'A'],
                [99, 67, 'B'],
            ],
            pack=True,
        ),
        plot=ui.plot(marks=[ui.mark(
            type='point',
            x='=x_value', x_title='My X Value',
            y='=y_value', y_title='My Y Value',
            color='=category', shape='circle', color_range="blue red", color_domain=["A", "B"]
        )])
    )
```
![image](https://user-images.githubusercontent.com/4793561/155635118-4487f9fe-355a-4ca6-8cf3-c6c27bb334d2.png)

#### Describe the solution you'd like
The `color_range` and `color_domain` features to be considered a "mapping" with a respected order.

#### Describe alternatives you've considered
Require the user to order their data appropriately. If we go with this we should add more documentation around it, and explicitly add a plot example that uses `color_domain`

**Comment by lo5** [comment]
> the categories don't have the colors we would expect.

In the plot above, A is blue, and B is red. The legend items are in the wrong order, which is odd, but the mapping is correct.

**Comment by mtanco** [comment]
You are totally right, I confused myself with the comment in #1267 where I first noticed this - it's just that the key is in an unexpected order.
